### PR TITLE
Initial Changes to use data bound constructor.

### PR DESCRIPTION
**Pull request by nbhoski**
* Used @databound constructors for each artifact class.
* Reduced scope of constant valuables to Artifact specific class.
* Single quote logic is scoped to Inputargs() method of builder class.

### CODE REVIEW

File: src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java [hunk 1]
```diff
@@ -247,149 +254,141 @@ private String getInputArguments() {
...
         @Override
         public boolean getDefault() {
-            
             return false;
         }
 
         @Override
         public String getFilePath() {
-            
             return null;
```
Comment (acampbel) [inline src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java:360]: Suggest returning "" instead of null

Comment (nbhoski) [reply src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java:360]: This has to be NULL due to default value in config.jelly  file . UI shows the default value only if the getter of that field is NULL. as we will assign Nullartifact by default after first save operation default values will not be populated on UI as NullArtifact returns  empty string.

File: src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java [hunk 2]
```diff
@@ -246,212 +253,142 @@ private String getInputArguments() {
...
      */
-    public static class PdfArtifact extends Artifact {
-        private String pdfReportFilePath;
+    public static class PdfArtifact extends AbstractArtifactImpl {
 
-        @DataBoundConstructor
-        public PdfArtifact() {
-           
-        }
+        static final String PDF_REPORT_PATH = "PDFReportPath";
```
Comment (acampbel) [inline src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java:258]: why not make all these private static final?

Comment (nbhoski) [reply src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java:258]: Done.

**Pull request merged**